src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -120,23 +142,40 @@ export const WorkspacePermissionSettingInput = ({
   );
 
   const handlePermissionModeOptionChange = useCallback(
-    (id: string) => {
-      if (optionIdToWorkspacePermissionModesMap[id]) {
-        onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+    (option: Array<EuiComboBoxOptionOption<any>>) => {
+      if (option.length > 0) {
+        const id = option[0].value;
+        if (optionIdToWorkspacePermissionModesMap[id]) {
+          onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+        }
       }
     },
     [index, onPermissionModesChange]
   );
 
+  const handleTypeChange = useCallback(
+    (options: Array<EuiSelectableOption<any>>) => {
+      for (const option of options) {
+        if (option.checked === 'on') {
+          onTypeChange(option.value, index);
+          setIsTypeListOpen(false);
+          return;
+        }
+      }
+    },
+    [index, onTypeChange]
+  );
```

Comment:
We need a unit test for this function.

Reply:
It's added here, but it doesn't seem to work src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.test.tsx:117

Reply:
The root cause should be missing `await` for `waitFor`. The `waitFor` return a promise. The test case will be completed if doesn't await the promise object.